doc/specs/stdlib_hash_maps.md

```diff
@@ -0,0 +1,3112 @@
+`other_type`, and associated procedures, for storing and manipulating
+keys and their associated data.
+
+### The `stdlib_hashmap_wrappers` constant, `INT_HASH`
+
+The constant `INT_HASH` is used to define the integer kind value for
+the returned hash codes and variables used to access them. It
+currently has the value, `INT32`.
+
+### The `stdlib_hashmap_wrappers` module derived types
+
+The `stdlib_hashmap_wrappers` module defines two derived types:
+`key_type`, and `other_type`. The `key_type` is intended to be used
+for the search keys of hash tables.  The `other_type` is intended to
+store additional data associated with a key. Both types are
+opaque. Their current representations are as follows
+
+```fortran
+    type :: key_type
+        private
+        integer(int8), allocatable :: value(:)
+    end type key_type
+
+    type :: other_type
+        private
+        integer(int8), allocatable :: value(:)
+    end type other_type
+```
+
+The  module also defines seven procedures for those types: `copy_key`,
+`copy_other`, `equal_keys`, `free_key`, `free_other`, `get`, and `set`
+for use by the hash maps to manipulate or inquire of components of
+those types.
+
+### Table of `stdlib_hashmap_wrappers` procedures
+
+The  `stdlib_hashmap_wrappers` module provides procedures in
+several categories: procedures to manipulate data of the `key_type`;
+procedures to manipulate data of the `other_type`, and 32 bit hash
+functions for keys. The procedures in each category are listed below.
+
+Procedures to manipulate `key_type` data:
+
+* `copy_key( key_in, key_out )` - Copies the contents of the key,
+  `key_in`, to contents of the key, `key_out`.
+
+* `equal_keys( key1, key2 )` - compares two keys for equality. 
+
+* `get( key, value )` - extracts the contents of key into value, an
+  `int8` array or character string.
+
+* `free_key( key )` - frees the memory in key.
+
+* `set( key, value )` - sets the content of key to value.
+
+Procedures to manipulate `other_type` data:
+
+* `copy_other( other_in, other_out )` - Copies the contents of the
+  other data, `other_in`, to the contents of the other data,
+  `other_out`.
+
+* `get( other, value )` - extracts the contents of other into value, an 
+  `int8` array or character string. 
+
+* `set( other, value )` - sets to content of other to value.
+
+* `free_other( other )` - frees the memory in other.
+
+Procedures to hash keys to 32 bit integers:
+
+* `fnv_1_hasher( key )` - hashes a key using the FNV-1 algorithm.
+
+* `fnv_1a_hasher( key )` - hashes a key using the FNV-1a algorithm.
+
+* `seeded_nmhash32_hasher( key )` - hashes a key using the nmhash32
+  algorithm.
+
+* `seeded_nmhash32x_hasher( key )` - hashes a key using the nmhash32x
+  algorithm.
+
+* `seeded_water_hasher( key )` - hashes a key using the waterhash
+  algorithm.
+
+### Specifications of the `stdlib_hashmap_wrappers` procedures
+
+#### `copy_key` - Returns a copy of the key
```

Comment:
could it overload "="?

Reply:
Yes it could. It would require exchanging argument locations and changing all references to `copy_key` in the code. The same could be done for `copy_other`.